### PR TITLE
Basic CEF Integration

### PR DESCRIPTION
**Pull request by koraa**
This branch contains an pre-alpha integration of CEF. It is always compiled into inexor, but not used at runtime unless it is explicitly activated.

In addition to that, this pull request also adds some generic utilities, some exception handling facilities and so-called Subsystems, which are parts of inexor that can be started, stopped and re-stared while inexor is running.
CEF is such a subsystem and RPC too.

Please note that the CEF can not at all be considered stable at the moment; it is integrated that it should not do harm unless activated. There are quite a few issues left, which are mentioned in their own tickets.

RPC and CEF can now be started/stopped by using the normal sauerbraten console:

```
subsystem_start cef
subsystem_stop cef

subsystem_start rpc
subsystem_stop rpc

cef_menu 1 # To start the menu
```

### CODE REVIEW

**Comment by koraa** [comment]
@a-teammate Could you please check whether this still compiles with VS?

**Comment by Croydon** [comment]
How to enable CEF?

**Comment by koraa** [comment]
(the build failed because the API dir upload took too long; the rest works and this can be merged)

File: src/ui/cefsettings.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	    	ignore_certificate_errors = true;
+	        multi_threaded_message_loop = false;
+	        windowless_rendering_enabled = true;
+	        command_line_args_disabled = true;
+	        log_severity = LOGSEVERITY_INFO;
+	        single_process = true;
+	        remote_debugging_port = 13012;
+	        char base_path[FILENAME_MAX];
+	        GetCurrentDir(base_path, sizeof(base_path));
+	        CefString(&resources_dir_path).FromString(std::string(base_path) + "/src/libraries/cef3/Resources");
```
Comment (a-teammate) [inline src/ui/cefsettings.h]: todo?

File: web/common.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+body {
```
Comment (a-teammate) [inline web/common.css]: maybe we could put the web/ folder to into the data repo? (so content guys can work on that)

Comment (Croydon) [inline web/common.css]: I support this strongly :)

Comment (koraa) [inline web/common.css]: web/ is not going to contain a lot of data. It is going to contain hardcore code for the web interface: Angular bindings; rest api clients;

**Comment by koraa** [comment]
The corrections for win have been squashed; I will be merging some when this evening

**Pull request merged**